test/src/e2e_vm_tests/test_programs/should_pass/language/main_args/main_args_mutation/test.toml

```diff
@@ -0,0 +1,4 @@
+category = "fail"
```

Comment:
Should test be moved to `should_fail`?

Reply:
But it's so good that all these tests related to this feature are in one place. Do we even need the `should_fail`/`should_pass` separation? Separating by feature seems more reasonable.

Reply:
But it's so good that all these tests related to this feature are in one place. Do we even need the `should_fail`/`should_pass` separation? Separating by feature seems more reasonable.

Reply:
I'm okay with unifying the folders, `should_fail` and `should_pass` are remnants of a pre-test.toml world. But yeah this PR should follow current practices and we can do a sweeping update afterwards.

Reply:
Created this issue for discussion: https://github.com/FuelLabs/sway/issues/2690